pkg/resource/edit/operations_test.go

```diff
@@ -91,6 +91,55 @@ func TestDeletion(t *testing.T) {
 	assert.Equal(t, []*resource.State{pA, a, c}, snap.Resources)
 }
 
+func TestDeletingDuplicateDependencies(t *testing.T) {
+	t.Parallel()
+
+	pA := NewProviderResource("a", "p1", "0")
+	a := NewResource("a", pA)
+
+	// Create duplicate resources.
+	b1 := NewResource("b", pA)
+	b2 := NewResource("b", pA)
+	b3 := NewResource("b", pA)
+
+	// ensure b1, b2, and b3 must have the same URN.
+	bURN := b1.URN
+	assert.Equal(t, bURN, b1.URN)
+	assert.Equal(t, bURN, b2.URN)
+	assert.Equal(t, bURN, b3.URN)
+
+	// c exists to check behavior on b's dependents.
+	c := NewResource("c", pA, bURN)
+
+	t.Run("do-target-dependents", func(t *testing.T) {
+		t.Parallel()
+		snap := NewSnapshot([]*resource.State{
+			pA, a, b1, b2, b3, c,
+		})
+
+		err := DeleteResource(snap, b1, nil, true /* targetDependents */)
+		require.NoError(t, err)
+
+		assert.Equal(t, []*resource.State{
+			pA, a, b2, b3, c,
+		}, snap.Resources)
```

Comment:
Use of assert.Equal here ends up not validating the reference-based deletion because assert.Equal does a deep equals.

If you replace the assert.Equal LHS with `{pA, a, b1, b3, c}`, it will still pass because b2 deep equals b1.

To verify the new behavior correctly, we need to do a reference match.
testify does not include a referential matching helper
so the simplest way to verify this might be to loop through the slice and do `==` comparisons.

```go
// assert.Equals or assert.NotContains
// are not enough because b1 deep equals b2 and b3.
for _, s := range snap.Resources {
  assert.False(t, s == b1)
}
```